src/tests/java/htsjdk/samtools/CRAMEdgeCasesTest.java

```diff
@@ -33,6 +34,25 @@ public void testUnsorted() throws IOException {
         testRecords(records, records.iterator().next().getReadBases());
     }
 
+    // int test for CRAMException
+    // testing for a contig found in the reads but not in the reference
+    @Test
+    public void testContigNotFoundInRef() throws IOException {
+        final File CRAMFile = new File("testdata/htsjdk/samtools/cram/test.cram");
+        final File refFile = new File("testdata/htsjdk/samtools/cram/test.fa");
+        final ReferenceSource refSource = new ReferenceSource(refFile);
+        final CRAMIterator iterator = new CRAMIterator(new FileInputStream(CRAMFile), refSource);
+        try {
+            while (iterator.hasNext()) {
+                iterator.next();
+            }
+        }
+        // the iterator is wrapping our exception; ensuring that it was actually a CRAMexception
+        catch (RuntimeEOFException e) {
+            Assert.assertEquals(e.getCause().getClass(), CRAMException.class);
```

Comment:
You also need to keep track of whether you got any exception at all, and assert that you did at the end of this test (since you can't use the built-in `expectedExceptions` here due to the wrapping.